fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/catalog/FlinkCatalog.java

```diff
@@ -263,6 +268,48 @@ public CatalogBaseTable getTable(ObjectPath objectPath)
                                             tableName.split("\\" + LAKE_TABLE_SPLITTER)[0])));
                 }
                 return getLakeTable(objectPath.getDatabaseName(), tableName);
+            } else if ((tableName.contains(CHANGELOG_TABLE_SPLITTER))) {
+                String baseTableName = tableName.split("\\" + CHANGELOG_TABLE_SPLITTER)[0];
+                TablePath baseTablePath = TablePath.of(objectPath.getDatabaseName(), baseTableName);
+                tableInfo = admin.getTableInfo(baseTablePath).get();
+                if (!tableInfo.hasPrimaryKey()) {
+                    throw new UnsupportedOperationException(
+                            String.format(
+                                    "\"Table %s has no primary key, only primary key tables support changelog.",
+                                    baseTableName));
+                }
+                CatalogTable originalTable = FlinkConversions.toFlinkTable(tableInfo);
+                Schema originalSchema = originalTable.getUnresolvedSchema();
+                List<Schema.UnresolvedColumn> newColumns =
+                        new ArrayList<>(originalSchema.getColumns());
+                newColumns.add(
+                        new Schema.UnresolvedPhysicalColumn("_change_type", DataTypes.STRING()));
+                newColumns.add(
+                        new Schema.UnresolvedPhysicalColumn("_log_offset", DataTypes.BIGINT()));
+                newColumns.add(
+                        new Schema.UnresolvedPhysicalColumn(
+                                "_commit_timestamp", DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE()));
+                Schema changeLogSchema =
+                        Schema.newBuilder()
+                                .fromColumns(newColumns)
```

Comment:
we can call `column(..).withComment(..)` here to make the code clean and attach addition column comments.